File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""
    Rectangle class module
"""


class Rectangle:
    """
    Rectangle class

    """

    def __init__(self, width=0, height=0):
        """
        Initialising a rectangle class.

        """
        self.width = width
        self.height = height

    @property
    def width(self):
        """ Getter function for private variable, width """
        return self.__width

    @width.setter
    def width(self, value):
        """
        Setter function for width.

        """
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        else:
            self.__width = value

    @property
    def height(self):
        """ Getter function for height """
        return self.__height

    @height.setter
    def height(self, value):
        """Setter function for private variable, height.

        Args:
            self: arg
            value: height
        """
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        else:
            self.__height = value

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
// class Rectangle
class Rectangle {
  constructor (w, h) {
    if ((w = parseInt(w)) > 0 &&
        (h = parseInt(h)) > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let q = 0; q < this.height; q++) {
      console.log('X'.repeat(this.width));
    }
  }
}
module.exports = Rectangle;
